apps/emqx_resource/src/emqx_resource_manager.erl

```diff
@@ -122,7 +122,10 @@ create_and_return_data(MgrId, ResId, Group, ResourceType, Config, Opts) ->
 -define(START_TIMEOUT, 5000).
 
 %% @doc Create a resource_manager and wait until it is running
-create(MgrId, ResId, Group, ResourceType, Config, Opts) ->
+create(MgrId, ResId, Group, ResourceType, Config, Opts0) ->
+    Enabled = maps:get(enable, Config, true),
```

Comment:
It is good, but beaks the design that emqx-resource never depends on the details of the `Config`. What the emqx-resource knows are the `ResId`, `ResType` and`Opts`, but the `Config` should be transparent to the `emqx-resource` level.

Reply:
In authz, we changed the opt `start_after_create` to false [here](https://github.com/emqx/emqx/blob/6bbb5edb536957425f11fc7f7ce38032cbf97208/apps/emqx_authz/src/emqx_authz_utils.erl#L43), and then start the resource only when enabled [here](https://github.com/emqx/emqx/blob/6bbb5edb536957425f11fc7f7ce38032cbf97208/apps/emqx_authz/src/emqx_authz_utils.erl#L62)